### PR TITLE
Reverse a String in Go

### LINKED ISSUE #538: Add Reverse a String in Go
If this is your first time contributing, create the solution file in the appropriate directory and add the name of the program to the Sample Programs section of the language README. Then, create a new issue for its article like #30. Link this issue in the README and create a pull request when you're done,

### PR DESCRIPTION
**Pull request by riddhik84**
Congrats on taking the first step to contributing to the [Sample Programs in Every Language][1] repository!

For simplicity, please make sure that your pull request includes one and only one sample program and language.

## Please Complete the Following

- [ ] I would **NOT** like credit for my contribution in the article (credit will be included otherwise)
- [ ] I plan to write the article for this code snippet
- [x] I named the pull request using `<Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [x] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [x] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])
- [ ] I fixed #538 

## Notes

Feel free to put whatever you want here.

[1]: https://therenegadecoder.com/code/sample-programs-in-every-language/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/CONTRIBUTING.md#plagiarism

### CODE REVIEW

File: archive/g/go/string-reverse.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+func reverse_string(str string) string {
+	chars := []rune(str)
+	for i, j := 0, len(chars)-1; i < j; i, j = i+1, j-1 {
+		chars[i], chars[j] = chars[j], chars[i]
+	}
+	return string(chars)
+}
+
+func main() {
+	fmt.Printf("%v\n", reverse_string("hacktoberfest"))
```
Comment (jrg94) [inline archive/g/go/string-reverse.go]: Can you pull the string from the command line?

Comment (jrg94) [reply archive/g/go/string-reverse.go]: Also, can you verify that there's a string on the command line to grab. For example:

```
if args.length > 0:
    // print reversed string
else:
    // print error/usage instructions
```

Comment (riddhik84) [reply archive/g/go/string-reverse.go]: I will update it.

Comment (jrg94) [reply archive/g/go/string-reverse.go]: Hey @riddhik84! Did you ever get around to this?

Comment (riddhik84) [reply archive/g/go/string-reverse.go]: Working on it. I will update the code today!

**Comment by riddhik84** [comment]
Updated program. Please check.

File: archive/g/go/string-reverse.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	for i, j := 0, len(chars)-1; i < j; i, j = i+1, j-1 {
+		chars[i], chars[j] = chars[j], chars[i]
+	}
+	return string(chars)
+}
+
+func main() {
+	// Check the command line args length
+	var argslen = len(os.Args)
+	//fmt.Println(argslen)
```
Comment (chrboe) [inline archive/g/go/string-reverse.go]: No commented out code please

File: archive/g/go/string-reverse.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	
+	// If the length is 2, one command line arg exist
+	if argslen == 2 {
+		input := os.Args[1]
+		fmt.Println("Input: " , input)
+		fmt.Printf("Reverse: %v\n", reverse_string(input))
```
Comment (chrboe) [inline archive/g/go/string-reverse.go]: For consistency, can we update this to output only the reversed string? (and not the input or "Reverse:")

Comment (jrg94) [reply archive/g/go/string-reverse.go]: Good point! I must not have noticed this the first time.

**Comment by riddhik84** [comment]
Updated!

**Pull request merged**